integration-tests/main/src/test/java/io/quarkus/it/main/MockTestCase.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkus.it.main;
+
+import javax.enterprise.inject.spi.CDI;
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.it.mock.MockableBean1;
+import io.quarkus.it.mock.MockableBean2;
+import io.quarkus.test.junit.MockSupport;
+import io.quarkus.test.junit.QuarkusMock;
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+public class MockTestCase {
+
+    protected static final String HI = "Hi ";
+    protected static final String BONJOUR = "Bonjour ";
+
+    @Inject
+    MockableBean1 mockableBean1;
+
+    @Inject
+    MockableBean2 mockableBean2;
+
+    @BeforeAll
+    public static void setup() {
+        QuarkusMock.installMock(CDI.current().select(MockableBean1.class).get(), new HiGreeter());
+    }
+
+    @Test
+    public void testBeforeAll() {
+        Assertions.assertEquals("Hi Stuart", mockableBean1.greet("Stuart"));
+        Assertions.assertEquals("Hello Stuart", mockableBean2.greet("Stuart"));
+    }
+
+    @Test
+    public void testPerTestMock() {
+        MockSupport.installMock(mockableBean2, new BonourGreeter());
```

Comment:
`MockSupport` is not part of the user API, or? In fact, it could be package-private...

Reply:
Yea, it should be. Originally this was in the common module and I did not update it.